src/lib/notification_service/formatter_provider.py

```diff
@@ -0,0 +1,24 @@
+from .formatter import HtmlFormatter
+
+
+class FormatterProvider:
+    """Formatter factory."""
+
+    def __init__(self):
+        self._formatters = {}
+
+    def register_sender(self, delivery_method, sender):
```

Comment:
Is _register_sender_ going to be called anywhere else besides this class? If not, why is it not preferable to initialize the dictionary of formatters in the constructor? The reason I'm asking is because I'm doing kind of the same with glue/stepfunctions event parsers :)